src/FixtureBuilder/Denormalizer/Fixture/Chainable/NullRangeNameReferenceDenormalizer.php

```diff
@@ -0,0 +1,193 @@
+    const REGEX = '/.+\{(?<expression>@(?<name>([A-Za-z0-9-_]+))(?<flag>(\*+))?)\}/';
+
+    /**
+     * @var FlagParserInterface|null
+     */
+    private $flagParser;
+
+    /**
+     * @var SpecificationsDenormalizerInterface
+     */
+    private $specsDenormalizer;
+
+    public function __construct(SpecificationsDenormalizerInterface $specsDenormalizer, FlagParserInterface $parser = null)
+    {
+        $this->specsDenormalizer = $specsDenormalizer;
+        $this->flagParser = $parser;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function withFlagParser(FlagParserInterface $parser): self
+    {
+        return new self($this->specsDenormalizer, $parser);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function canDenormalize(string $name, array &$matches = []): bool
+    {
+        return 1 === preg_match(self::REGEX, $name, $matches);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function denormalize(
+        FixtureBag $builtFixtures,
+        string $className,
+        string $fixtureId,
+        array $specs,
+        FlagBag $flags
+    ): FixtureBag {
+        $matches = [];
+        if (false === $this->canDenormalize($fixtureId, $matches)) {
+            throw LogicExceptionFactory::createForCannotDenormalizerForChainableFixtureBuilderDenormalizer(__METHOD__);
+        }
+
+        $referencedName = $matches['name'];
+        $allFlag = ($matches['flag'] ?? null) === '*';
+
+        $fixtureIds = $this->buildReferencedValues($builtFixtures, $referencedName, $allFlag);
+
+        $fixtureIdPrefix = $this->determineFixtureIdPrefix($fixtureId);
+
+        foreach ($fixtureIds as $referencedFixtureId => $valueForCurrent) {
+            $builtFixtures = $builtFixtures->with(
+                $this->buildFixture(
+                    $fixtureIdPrefix . $referencedFixtureId,
+                    $className,
+                    $specs,
+                    $flags,
+                    $valueForCurrent
+                )
+            );
+        }
+
+        return $builtFixtures;
+    }
+
+    /**
+     * @param FixtureBag $builtFixtures
+     * @param string     $referencedName
+     * @param bool       $allFlag
+     *
+     * @return array
+     */
+    private function buildReferencedValues(
+        FixtureBag $builtFixtures,
+        string $referencedName,
+        bool $allFlag
+    ): array {
+        if (false === $allFlag) {
+            $fixture = $builtFixtures->get($referencedName);
+
+            return [$referencedName => $fixture];
+        }
+
+        $matchedFixtures = array_filter(
+            $builtFixtures->toArray(),
+            function (string $referenceName) use ($referencedName) {
+                return strpos($referenceName, $referencedName) === 0;
+            },
+            ARRAY_FILTER_USE_KEY
+        );
+
+        return $matchedFixtures;
+    }
+
+    /**
+     * @param string $fixtureId
+     *
+     * @return string
+     * @throws \LogicException
```

Comment:
in general it's unnecessary to document `\LogicException`s: they are sign of an incorrect config requiring manual fix, i.e. it's not an error you should ever encounter at runtime (unlike `\RuntimeException`) and account for

Reply:
That was the ******** phpstorm bug with exceptions :-(